ui/scripts/launch_replay.py

```diff
@@ -0,0 +1,146 @@
+    help="""Path to the replay file. The file needs to either be a .JSON or a
+              .JSON.gz file. The file is generated if the api_recording flag is
+              set in the Cloe stackfile""",
+)
+@click.option(
+    "-h",
+    "--host",
+    default="localhost",
+    help="IP adress of host where ui and server run. (Default is localhost)",
+)
+@click.option("--cloe-ui-image", required=True, help="Cloe-ui docker image.")
+@click.option(
+    "--webserver-image", required=True, help="Cloe-ui webserver docker image."
+)
+def main(path: str, host: str, cloe_ui_image: str, webserver_image: str):
+    """Replay simulation data with the Cloe web-ui.
+    Docker containers which are used:
+        - cloe-ui
+        - cloe-ui-webserver
+    If there are no containers running, this script will start both.
+    If the images are not found locally, they will be pulled from the repository.
+    """
+
+    data_file_path = Path(path).resolve()
+    dirname = data_file_path.parent.name
+    return_code = launch_cloe_ui_webserver(data_file_path, dirname, webserver_image)
+    if return_code == 0:
+        print(
+            "-------------  Launched Webserver at http://localhost:4000 ------------- "
+        )
+    else:
+        print(f"Failed to launch webserver: {return_code}")
+        quit()
+
+    return_code = launch_cloe_ui(cloe_ui_image)
+    if return_code == 0:
+        print("------------- Launched Cloe UI at http://localhost:5000 ------------- ")
+    else:
+        print(f"Failed to launch web ui: {return_code}")
+        quit()
+
+    sleep(3)
+    open_browser(host, dirname, data_file_path.name)
+
+
+def open_browser(host, dir, filename):
+    url = f"http://{host}:5000?id={dir}&name={filename}&host={host}"
+    webbrowser.open_new(url)
+
+
+def launch_cloe_ui(docker_image):
+    docker_container_name = "cloe-ui"
+    # Stop cloe ui container if it is running.
+    stop_container(docker_container_name)
+    # Pull latest image from DTR.
+    run_process(["docker", "pull", docker_image])
+
+    commands = []
+    commands.extend(
+        [
+            "docker",
+            "run",
+            "--rm",
+            "-d",
+            f"--name={docker_container_name}",
+            "-p",
+            "5000:5000",
+            docker_image,
+        ]
+    )
+
+    return run_process(commands)
+
+
+def stop_container(docker_container_name):
+    docker_env = docker.from_env()
+    running_container = docker_env.containers.list(
+        filters={"status": "running", "name": docker_container_name}
+    )
+    for container in running_container:
+        if container.name == docker_container_name:
+            container.stop()
+
+
+def launch_cloe_ui_webserver(mounting_path, dir, docker_image):
+    docker_container_name = "cloe-ui-webserver"
+
+    # Stop webserver container if it is running.
+    stop_container(docker_container_name)
+    # Pull latest image from DTR.
+    run_process(["docker", "pull", docker_image])
```

Comment:
By default "docker run" will pull a missing image. Whereas "docker pull" will pull unconditionally. Which might cause issues in case the image is locally already present but not available on the DTR.